tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+                // the context.
+                //                assert!(key.to_fingerprint_string().contains(&*FINGERPRINTS[pubkeyvalid1]));
+
+                let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ssh_dir, testnameone));
+
+                assert!(!authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(pubkeyvalid1);
+                assert_has_keys(ssh_dir, svec, true);
+            }
+        }
+    }
+
+    #[test]
+    fn test_del_two() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let pubkeyvalid2 = "valid2";
+        let testnameone = "one";
+        let testnametwo = "two";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        let assert_keys = ["valid1", "valid2"].to_vec();
+
+        // "update-ssh-keys --add two valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnametwo,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        // "update-ssh-keys --delete two valid2.pub"
+
+        match del_one_ssh_key(ssh_dir, pubkeyvalid2, testnametwo) {
+            Ok(ake) => {
+                println!("update_ssh_keys --delete passed");
+
+                for key in ake {
+                    if let AuthorizedKeyEntry::Valid {
+                        ref key,
+                    } = key
+                    {
+                        assert!(key.to_fingerprint_string().contains(&*FINGERPRINTS[pubkeyvalid2]));
+
+                        let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ssh_dir, testnametwo));
+
+                        assert!(!authkeyone.exists());
+
+                        let mut svec: Vec<&str> = Vec::new();
+                        svec.push(pubkeyvalid1);
+                        assert_has_keys(ssh_dir, svec, true);
+                    }
+                }
+            }
+            Err(err) => panic!("update_ssh_keys --delete failed {}", err),
+        }
+    }
+
+    fn run_test_disable(ctx: TestContext) {
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+
```

Comment:
I rather would like to see a call to `run_test_add_two(&ctx)` here than the duplication of the logic.

Reply:
Called `run_test_add_two(&ctx.ssh_dir)`.